repositories/src/test/scala/cromwell/pipeline/datastorage/dao/repository/RunRepositoryTest.scala

```diff
@@ -0,0 +1,77 @@
+package cromwell.pipeline.datastorage.dao.repository
+
+import com.dimafeng.testcontainers.{ ForAllTestContainer, PostgreSQLContainer }
+import com.typesafe.config.Config
+import cromwell.pipeline.datastorage.DatastorageModule
+import cromwell.pipeline.datastorage.dao.repository.utils.{ TestProjectUtils, TestRunUtils, TestUserUtils }
+import cromwell.pipeline.datastorage.dto.{ Project, Run, User }
+import cromwell.pipeline.utils.{ ApplicationConfig, TestContainersUtils }
+import org.scalatest.{ AsyncWordSpec, BeforeAndAfterAll, Matchers }
+
+class RunRepositoryTest extends AsyncWordSpec with Matchers with BeforeAndAfterAll with ForAllTestContainer {
+
+  private lazy val config: Config = TestContainersUtils.getConfigForPgContainer(container)
+  private lazy val datastorageModule: DatastorageModule = new DatastorageModule(ApplicationConfig.load(config))
+  override val container: PostgreSQLContainer = TestContainersUtils.getPostgreSQLContainer()
+  private val dummyUser: User = TestUserUtils.getDummyUser()
+  private val dummyProject: Project = TestProjectUtils.getDummyProject(ownerId = dummyUser.userId)
+  private val dummyRun: Run = TestRunUtils.getDummyRun(userId = dummyUser.userId, projectId = dummyProject.projectId)
+
+  override protected def beforeAll(): Unit = {
+    super.beforeAll()
+    datastorageModule.pipelineDatabaseEngine.updateSchema()
+  }
+
+  import datastorageModule.{ projectRepository, runRepository, userRepository }
+  "RunRepository" when {
+
+    "getRunById" should {
+
+      "find newly added run by id" taggedAs Dao in {
+        val addUserFuture = userRepository.addUser(dummyUser)
+        val addProjectFuture = projectRepository.addProject(dummyProject)
+        val result = for {
+          _ <- addUserFuture
+          _ <- addProjectFuture
+          _ <- runRepository.addRun(dummyRun)
+          getById <- runRepository.getRunById(dummyRun.runId)
+        } yield getById
+        result.map(optRun => optRun shouldEqual Some(dummyRun))
+      }
+    }
+
+    "update run by id" should {
+
+      "update status, time start, time end, result and cmwlWorkflowId" taggedAs Dao in {
+        val updatedRun =
+          dummyRun.copy(
+            status = "new-status",
+            timeStart = TestRunUtils.getDummyTimestamp,
+            timeEnd = TestRunUtils.getDummyTimestamp,
+            results = "new-result",
+            cmwlWorkflowId = "new-cmwlWorkflowId"
+          )
+
+        val result = for {
+          _ <- runRepository.updateRun(updatedRun)
+          getById <- runRepository.getRunById(dummyRun.runId)
+        } yield getById
+
+        result.map(optRun => optRun shouldEqual Some(updatedRun))
+      }
+    }
+
+    "delete run" should {
+
+      "delete run by id" taggedAs Dao in {
+        val result = for {
+          _ <- runRepository.deleteRunById(dummyRun.runId)
+          getById <- runRepository.getRunById(dummyRun.runId)
```

Comment:
Same problem as with `update`.
You never inserted `Run` in this test, therefore this test should always pass even if `deleteRunById` does nothing.